xarray/plot/utils.py

```diff
@@ -221,8 +221,14 @@ def _infer_xy_labels(darray, x, y):
         if darray.ndim != 2:
             raise ValueError('DataArray must be 2d')
         y, x = darray.dims
-    elif x is None or y is None:
-        raise ValueError('cannot supply only one of x and y')
+    elif x is None:
+        if y not in darray.dims:
+            raise ValueError('y must be a coordinate variable')
```

Comment:
> Should we check for x and y in darray.coords then, instead of darray.dims?

No, I don't think this would work. 

Since we will probably land on the name "coordinate variable" for the "dimension" coords, I guess we can go back to @shoyer 's initial suggestion, or change the error message to:

```y must be a dimension name if x is not supplied```

which has the advantage to cover all cases (also when there are unindexed dimensions).